lib/datadog/appsec/contrib/rack/gateway/watcher.rb

```diff
@@ -65,10 +65,10 @@ def watch_request(gateway = Instrumentation.gateway)
               end
 
               def watch_response(gateway = Instrumentation.gateway)
-                gateway.watch('rack.response', :appsec) do |stack, response|
+                gateway.watch('rack.response', :appsec) do |stack, gateway_response|
                   block = false
                   event = nil
-                  waf_context = response.instance_eval { @waf_context }
+                  waf_context = AppSec::Processor.active_context
```

Comment:
This can be problematic, for instance because of [stuff like this](https://github.com/rails/rails/blob/1512cf2ba578282c898b8eb68ac401ea5243b7b5/actionpack/lib/action_controller/metal/live.rb#L261-L263).

That's why I favoured accessing the context from a known source. Now that we have objects in our control we can have the active context accessible without a `instance_eval` hack.